entr.c

```diff
@@ -155,15 +156,22 @@ main(int argc, char *argv[]) {
 		err(1, "Failed to set SIGCHLD handler");
 
 	getrlimit(RLIMIT_NOFILE, &rl);
+
 #if defined(_LINUX_PORT)
-	rl.rlim_cur = (rlim_t)fs_sysctl(INOTIFY_MAX_USER_WATCHES);
+	max_watches = (rlim_t)fs_sysctl(INOTIFY_MAX_USER_WATCHES);
+	if(max_watches > 0) {
+		rl.rlim_cur = max_watches;
+		goto rlim_set;
```

Comment:
This `goto` isn't needed; the lines following are only called if the platform is not Linux

Reply:
Oops, I forgot to modify the directives to unconditionally compile what had been in the #else

I've updated the code to what I had intended.  The goto is required because we want to use the original method when the new method fails, and jump over it when it succeeds.